### PR TITLE
Preserve full request history for continuous batching logits processors

### PR DESCRIPTION
**Pull request by Robertboy18**
[CPU CI](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=48468) [GPU run-slow](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=48468)

I was looking at how logits processors work with continuous batching and noticed that history-dependent processors were only receiving the current token instead of each request's prompt + generated history.

This preserves the full history for processors such as repetition penalty, no-repeat n-grams, sequence bias, forced BOS/EOS, and min length. The existing optimized temperature/top-k/top-p path remains unchanged.

I added focused parity tests and an end-to-end Qwen regression test. I can also attach the corresponding TorchLean/Lean formalization if useful :)

### CODE REVIEW

**Comment by Rocketknight1** [comment]
I think that's the first time we've ever been offered a Lean formalization of an issue, which is exciting! Do you have examples of the issues this is causing in practice, though? An example where one or more of these logit processors emits incorrect results in CB would be very helpful for triaging/assigning this!
